ci/L0_backend_vllm/metrics_test/vllm_metrics_test.py

```diff
@@ -0,0 +1,171 @@
+            "vllm:prompt_tokens_total": 0,
+            "vllm:generation_tokens_total": 0,
+        }
+
+        # Test vLLM metrics
+        self._test_vllm_model(
+            prompts=PROMPTS,
+            sampling_parameters=SAMPLING_PARAMETERS,
+            stream=False,
+            send_parameters_as_tensor=True,
+            model_name=self.vllm_model_name,
+        )
+        expected_metrics_dict["vllm:prompt_tokens_total"] = 18
+        expected_metrics_dict["vllm:generation_tokens_total"] = 48
+        print(get_metrics())
+        print(expected_metrics_dict)
+        self.assertEqual(get_metrics(), expected_metrics_dict)
+
+        self._test_vllm_model(
+            prompts=PROMPTS,
+            sampling_parameters=SAMPLING_PARAMETERS,
+            stream=False,
+            send_parameters_as_tensor=False,
+            model_name=self.vllm_model_name,
+        )
+        expected_metrics_dict["vllm:prompt_tokens_total"] = 36
+        expected_metrics_dict["vllm:generation_tokens_total"] = 96
+        self.assertEqual(get_metrics(), expected_metrics_dict)
+
+    def _test_vllm_model(
+        self,
+        prompts,
+        sampling_parameters,
+        stream,
+        send_parameters_as_tensor,
+        exclude_input_in_output=None,
+        expected_output=None,
+        model_name="vllm_opt",
+    ):
+        user_data = UserData()
+        number_of_vllm_reqs = len(prompts)
+
+        self.triton_client.start_stream(callback=partial(callback, user_data))
+        for i in range(number_of_vllm_reqs):
+            request_data = create_vllm_request(
+                prompts[i],
+                i,
+                stream,
+                sampling_parameters,
+                model_name,
+                send_parameters_as_tensor,
+                exclude_input_in_output=exclude_input_in_output,
+            )
+            self.triton_client.async_stream_infer(
+                model_name=model_name,
+                request_id=request_data["request_id"],
+                inputs=request_data["inputs"],
+                outputs=request_data["outputs"],
+                parameters=sampling_parameters,
+            )
+
+        for i in range(number_of_vllm_reqs):
+            result = user_data._completed_requests.get()
+            if type(result) is InferenceServerException:
+                print(result.message())
+            self.assertIsNot(type(result), InferenceServerException, str(result))
+
+            output = result.as_numpy("text_output")
+            self.assertIsNotNone(output, "`text_output` should not be None")
+            if expected_output is not None:
+                self.assertEqual(
+                    output,
+                    expected_output[i],
+                    'Actual and expected outputs do not match.\n \
+                                  Expected "{}" \n Actual:"{}"'.format(
+                        output, expected_output[i]
+                    ),
+                )
```

Comment:
Are we using `expected_output` in this test?

Reply:
Updated